File: client/src/services/supabaseClient.ts
import { createClient } from "@supabase/supabase-js";

const SUPABASE_URL = import.meta.env.VITE_SUPABASE_URL;
const SUPABASE_ANON_KEY = import.meta.env.VITE_SUPABASE_ANON_KEY;

if (!SUPABASE_URL || !SUPABASE_ANON_KEY) {
  throw new Error(
    "Missing VITE_SUPABASE_URL or VITE_SUPABASE_ANON_KEY. Copy client/.env.example to client/.env and fill them in from your Supabase project's API settings."
  );
}

/**
 * The anon key is safe to expose in client code — it's meant to be
 * public. Actual data access is restricted by the Row Level Security
 * policies defined in supabase/schema.sql, not by keeping this key
 * secret. Never put the service_role key here.
 */
export const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

File: voice-verification-service/supabase_client.py
"""
supabase_client.py — two deliberately different ways of talking to
Supabase from this service, matching two different trust levels:

- get_user_scoped_client(): makes requests AS the calling user (their
  own JWT), so Row Level Security applies exactly as it would for any
  normal client request. Used for enrollment — writing your OWN
  voiceprint is safe under ordinary RLS, no special access needed.

- get_admin_client(): uses the service_role key, which BYPASSES RLS
  entirely. Used ONLY inside the /verify endpoint's one legitimate
  cross-user read (checking a caller's audio against THEIR stored
  embedding, on behalf of the person receiving the call — see main.py
  for the full reasoning). Never used for writes in this service.
"""
import os
import subprocess

from dotenv import load_dotenv
from supabase import create_client, Client

# Loads variables from a .env file in this folder, if one exists.
# On Hugging Face Spaces (or any real deployment), you won't have a
# .env file — you'll set these as actual environment variables /
# repository secrets instead, and load_dotenv() simply does nothing in
# that case (no .env file to find), which is exactly the behavior you
# want: same code works both ways.
load_dotenv()

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_ANON_KEY = os.getenv("SUPABASE_ANON_KEY")
SUPABASE_SERVICE_ROLE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

if not SUPABASE_URL or not SUPABASE_ANON_KEY or not SUPABASE_SERVICE_ROLE_KEY:
    raise RuntimeError(
        "Missing SUPABASE_URL, SUPABASE_ANON_KEY, or SUPABASE_SERVICE_ROLE_KEY. "
        "Set them in this service's environment before starting."
    )


def get_user_scoped_client(user_jwt: str) -> Client:
    client = create_client(SUPABASE_URL, SUPABASE_ANON_KEY)
    client.postgrest.auth(user_jwt)
    return client


def get_admin_client() -> Client:
    return create_client(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY)


def convert_to_wav(input_path: str, output_path: str, sample_rate: int = 16000) -> None:
    """Browser MediaRecorder output (webm/opus) -> mono 16kHz WAV, which
    both Silero VAD and ECAPA-TDNN expect."""
    cmd = [
        "ffmpeg", "-y", "-i", input_path,
        "-ac", "1", "-ar", str(sample_rate),
        output_path,
    ]
    subprocess.run(cmd, check=True, capture_output=True)
